src/integration-test/java/io/aiven/connect/jdbc/pg/PgSqlSourceConnectorIT.java

```diff
@@ -0,0 +1,122 @@
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+import io.aiven.connect.jdbc.JdbcSourceConnector;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.com.fasterxml.jackson.core.JsonProcessingException;
+import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PgSqlSourceConnectorIT extends AbstractPgSqlAwareIT {
+
+    @BeforeEach
+    void createDbTable() throws Exception {
+        try (final var stm = pgConnection.createStatement()) {
+            stm.execute(
+                    String.format("CREATE TABLE %s ("
+                            + " id INT PRIMARY KEY NOT NULL,"
+                            + " json_value JSON NOT NULL,"
+                            + " jsonb_value JSONB NOT NULL"
+                            + " )", TEST_TOPIC_NAME));
+        }
+        final var insertSql =
+                "INSERT INTO " + TEST_TOPIC_NAME
+                        + "(id, json_value, jsonb_value) "
+                        + "VALUES(?, ?::json, ?::jsonb)";
+        try (final var stm = pgConnection.prepareStatement(insertSql)) {
+            var rowCounter = 0;
+            for (int i = 1; i < 4; i++) {
+                stm.setInt(1, i);
+                stm.setString(2, String.format("{\"json_value\": %s}", i));
+                stm.setString(3, String.format("{\"jsonb_value\": %s}", i));
+                rowCounter += stm.executeUpdate();
+            }
+            assert rowCounter == 3;
+        }
+    }
+
+    @Test
+    void pgSqlSupportTypes() throws Exception {
+        final var config = new HashMap<String, String>();
+        config.put("name", "pg-types-source-connector");
+        config.put("tasks.max", "1");
+        config.put("mode", "incrementing");
+        config.put("incrementing.column.name", "id");
+        config.put("connector.class", JdbcSourceConnector.class.getCanonicalName());
+        config.put("topic.prefix", "");
+        config.put("tables", "pg_source_table");
+        config.put("connection.url", pgSqlContainer.getJdbcUrl());
+        config.put("connection.user", pgSqlContainer.getUsername());
+        config.put("connection.password", pgSqlContainer.getPassword());
+        config.put("key.converter", "org.apache.kafka.connect.json.JsonConverter");
+        config.put("value.converter", "org.apache.kafka.connect.json.JsonConverter");
+
+        jdbcConnectService.createConnector(config);
+
+        Thread.sleep(OFFSET_FLUSH_INTERVAL_MS * 8);
```

Comment:
Maybe instead of waiting so long, `poll.interval.ms` and `table.poll.interval.ms` should be configured?

Reply:
Sounds good. Will add `table.poll.interval.ms` as test param